onnx/defs/tensor/defs.cc

```diff
@@ -3952,4 +3952,126 @@ ONNX_OPERATOR_SET_SCHEMA(
           return true;
         }));
 
+static const char* TensorScatter_ver24_doc = R"DOC(
+TensorScatter is a generic tensor update operation, motivated by the requirements for KV cache updates for Attention
+ops commonly found in LLMs. It is a functional operation that models an in-place update to a KV cache buffer.
+
+The past and present cache tensors have the same shape, with the sequence length dimension (indicated by the
+`axis` attribute) being max_sequence_length, so the sizes of these tensors do not need to grow between iterations.
+
+The optional `write_indices` input indicates the write index for each sample in the batch, assumed to be zero
+if not provided. During the prefill phase of attention, only the first two inputs are needed. During the decode
+phase, `write_indices` is also needed so that the incoming k and v can be appended after the last valid token
+for each sample in the batch.
```

Comment:
One more question: what should happen if length(updates) > length(cache) in the circular mode? Is this an error situation with undefined behavior?

Also: may be useful to add a paragraph here describing the mode attribute, and the requirements (like above).

Reply:
For kv cache use cases that will not be a valid scenario (even for circular mode), so I think we can just leave it as undefined behavior. The name of the dimensions also hint at it "sequence_length" and "max_sequence_length".

We did discuss the rare case where the sequence is larger than circular buffer size. There all tokens need to go through Attention before being truncated to cache size and written to buffer. That would require a different kind of graph.